tests/integration/docarray_v2/test_v2.py

```diff
@@ -0,0 +1,142 @@
+from typing import Optional
+
+import numpy as np
+from docarray import BaseDocument, DocumentArray
+from docarray.documents import Image
+from docarray.typing import AnyTensor, ImageUrl
+
+from jina import Deployment, Executor, Flow, requests
+
+
+def test_different_document_schema():
+    class Image(BaseDocument):
+        tensor: Optional[AnyTensor]
+        url: ImageUrl
+
+    class MyExec(Executor):
+        @requests(on='/foo')
+        def foo(self, docs: DocumentArray[Image], **kwargs) -> DocumentArray[Image]:
+            for doc in docs:
+                doc.tensor = doc.url.load()
+            return docs
+
+    with Flow().add(uses=MyExec) as f:
+        docs = f.post(
+            on='/foo',
+            inputs=DocumentArray[Image](
```

Comment:
is this necessary ?
I believe that input casting (with DocumentArray[Image]) should not be a requirement for v2

Reply:
I think it is needed. Otherwise how do you know how to open to proto and with which class to instantiate ?

Reply:
correct me if I'm wrong but it looks like `DocumentArray.to_protobuf` does not need to know about the schema to serialize to protobuf
<img width="507" alt="image" src="https://user-images.githubusercontent.com/15269265/214840186-78eee496-46f5-4d08-8a86-a0f0c2af9394.png">

Reply:
hmmm good point good point, but actually that might already work then to not precise the output schema. Though I believe we should enforce ppl putting an output schema. This might help for example Executor interoperability, i.e, I can check the output schema of Executor X to see if it is compatible with Executor Y

Reply:
for output schema yes I agree but I was referring to the input schema, and I don't think it is necessary to specify it

Reply:
then  I don't understand how do you do if you don't have the input schema to create the proper document array ? Morevoer same argument I would rather expliclty encourage ppl to put the schema

Reply:
lets handle this in the next PR if needed